### PR TITLE
[#12048] Create SQL Logic for Get User Actions

### PR DESCRIPTION
**Pull request by domlimm**
Part of #12048.

**Outline of Solution**

This PR has the following changes:
- Add db and user logic for get instructor and student
- Add unit and integration tests
- Update related entities

### CODE REVIEW

File: src/test/java/teammates/storage/sqlapi/UsersDbTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+public class UsersDbTest extends BaseTestCase {
+
+    private UsersDb usersDb = UsersDb.inst();
+
+    private MockedStatic<HibernateUtil> mockHibernateUtil;
+    private UUID userId;
+
+    @BeforeMethod
+    public void setUpMethod() {
+        userId = UUID.randomUUID();
```
Comment (samuelfangjw) [inline src/test/java/teammates/storage/sqlapi/UsersDbTest.java:33]: Not required.

**Review by samuelfangjw — CHANGES_REQUESTED** [review]

File: src/main/java/teammates/storage/sqlentity/User.java [hunk 2]
```diff
@@ -82,6 +86,14 @@ public void setAccount(Account account) {
         this.account = account;
     }
 
+    public String getCourseId() {
+        return courseId;
+    }
+
+    public void setCourseId(String courseId) {
+        this.courseId = courseId;
+    }
```
[the remark this answers is not in the document]
Comment (domlimm) [reply src/main/java/teammates/storage/sqlentity/User.java:95]: Yeah! Thanks for catching this samuel

File: src/it/java/teammates/it/storage/sqlapi/UsersDbIT.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        course = new Course("course-id", "course-name", Const.DEFAULT_TIME_ZONE, "institute");
+        coursesDb.createCourse(course);
+
+        Section section = new Section(course, "section-name");
+        HibernateUtil.persist(section);
+
+        Team team = new Team(section, "team-name");
+        HibernateUtil.persist(team);
+
+        generateInstructor();
```
[the remark this answers is not in the document]
Comment (domlimm) [reply src/it/java/teammates/it/storage/sqlapi/UsersDbIT.java:47]: Noted!

File: src/main/java/teammates/storage/sqlentity/User.java [hunk 4]
```diff
@@ -57,10 +60,11 @@ protected User() {
         // required by Hibernate
     }
 
-    public User(Course course, Team team, String name, String email) {
+    public User(Course course, String name, String email) {
         this.setId(UUID.randomUUID());
+        this.courseId = course.getId();
```
Comment (samuelfangjw) [inline src/main/java/teammates/storage/sqlentity/User.java:65]: this should be in the setter of course

Comment (domlimm) [reply src/main/java/teammates/storage/sqlentity/User.java:65]: In that case, we can leave Line 37 as it is?

Comment (samuelfangjw) [reply src/main/java/teammates/storage/sqlentity/User.java:65]: yes leave as is. idea is we should not be able to set course and courseId independently of each other.

Comment (domlimm) [reply src/main/java/teammates/storage/sqlentity/User.java:65]: I understand, thanks for clarifying!!

**Review by samuelfangjw — APPROVED** [review]
Some issues with unit tests, will be resolved at a later time.

**Pull request merged**